wrappers/src/marshalling.hpp

```diff
@@ -28,6 +28,15 @@
 
 namespace realm::binding {
 
+/// A struct used when marshaling of `MarshaledVector` cannot be
+/// compiled, e.g. for MSVC when returning a `MarshaledVector` from
+/// CPP directly, as compared to when nested within another struct.
+struct AnyMarshaledVector
```

Comment:
I'd probably call this `TypeErasedMarshaledVector` - it's a bit more verbose, but also more descriptive for the intended usage.

Reply:
Good idea 🙂